test/e2e/common_test.go

```diff
@@ -983,13 +983,26 @@ func populateCache(podman *PodmanTestIntegration) {
 // rmAll removes the directory and its content, when running rootless we use
 // podman unshare to prevent any subuid/gid problems
 func rmAll(podmanBin string, path string) {
+	// When using overlay, podman leaves a stray mount behind. This
+	// leak causes remote tests to take a loooooong time, which then
+	// causes Cirrus to time out. Unmount the stray.
+	overlayPath := path + "/root/overlay"
+	if _, err := os.Stat(overlayPath); err == nil {
+		umount := exec.Command("umount", overlayPath)
+		umount.Stdout = GinkgoWriter
+		umount.Stderr = GinkgoWriter
+		if err = umount.Run(); err != nil {
+			GinkgoWriter.Printf("Error umounting %s: %v\n", overlayPath, err)
+		}
+	}
```

Comment:
This only works as root and would always fail for rootless, no? And it would be much better to just call unix.Unmount() instead of fork/exec each time.
Second you could add `--storage-opt=skip_mount_home=true` which should prevent c/storage from creating this mount in the first place.

Reply:
I took a look at the logs and yes this fails and spam the log for each test:
```
umount: /tmp/ginkgo1538884863/root/overlay: not mounted.
Error umounting /tmp/ginkgo1538884863/root/overlay: exit status 1
```
https://api.cirrus-ci.com/v1/artifact/task/5247050451779584/html/int-podman-fedora-39%CE%B2-rootless-host-sqlite.log.html

Reply:
Thank you. My purpose was to make things work (eliminate the CI timeout).

Re: that option, would that make the e2e tests too divergent from a real-world environment?

Reply:
```
Error: vfs driver does not support skip_mount_home options
```

Reply:
I've fixed this code to (1) run only when root, and (2) use `unix.Unmount()`. Seems to work, so I've marked again as ready for review.

The `cp` flake still exists, saw one today.